src/browser/BrowserService.cpp

```diff
@@ -1093,15 +1093,26 @@ bool BrowserService::checkLegacySettings()
         return false;
     }
 
-    auto dialogResult =
-        MessageBox::warning(nullptr,
-                            tr("KeePassXC: Legacy browser integration settings detected"),
-                            tr("Your KeePassXC-Browser settings need to be moved into the database settings.\n"
-                               "This is necessary to maintain your current browser connections.\n"
-                               "Would you like to migrate your existing settings now?"),
-                            MessageBox::Yes | MessageBox::No);
-
-    return dialogResult == MessageBox::Yes;
+    QMessageBox warning;
+    warning.setIcon(QMessageBox::Warning);
+    warning.setWindowTitle(tr("KeePassXC: Legacy browser integration settings detected"));
+    warning.setText("Your KeePassXC-Browser settings need to be moved into the database settings.\n"
+                    "This is necessary to maintain your current browser connections.\n"
+                    "Would you like to migrate your existing settings now?");
+    auto noButton = warning.addButton(QMessageBox::No);
+    auto yesButton = warning.addButton(QMessageBox::Yes);
+    auto rememberButton = warning.addButton(tr("No. Don't ask again."), QMessageBox::ButtonRole::RejectRole);
```

Comment:
Couple things here. This should be using the custom `MessageBox` class. Also, instead of adding a dedicated button, use the `MessageBox::setCheckBox(...)` function to add a checkbox with the following text: `tr("Don't show this warning again")` which is already translated.

Reply:
The only reason I used this implementation is that `MessageBox` class' constuctor executes the popup right away and returns the the button pressed. There's no default constructor available. So maybe this class needs to be extended to hold a checkbox inside if configured so.

Reply:
Necessary changes has been made.

Reply:
Hmmm that is a good point

Reply:
I was mistaken when I looked at the code yesterday, the other example for a checkbox used the QMessageBox directly. However, I do like what you did so we can now convert the other example to using MessageBox. The benefit of MessageBox is you can set a "next answer" during GUI testing to dismiss the box correctly.